packages/plugin-core/src/workspace/workspaceActivator.ts

```diff
@@ -0,0 +1,661 @@
+      onSyncVaultsEnd: () => {
+        vscode.window.showInformationMessage(
+          "finish initializing remote vaults. reloading workspace"
+        );
+        // TODO: remove
+        setTimeout(VSCodeUtils.reloadWindow, 200);
+      },
+    });
+    if (didClone) {
+      return {
+        error: ErrorFactory.createInvalidStateError({
+          message: "could not initialize workspace",
+        }),
+      };
+    }
+    Logger.info({ ctx: `${ctx}:postWsServiceInitialize`, wsRoot });
+
+    // check for vaults with duplicates
+    const respNoDupVault = await checkNoDuplicateVaultNames(wsService.vaults);
+    if (!respNoDupVault) {
+      return {
+        error: ErrorFactory.createInvalidStateError({
+          message: "found duplicate vaults",
+        }),
+      };
+    }
+
+    // write new workspace version
+    wsService.writeMeta({ version: DendronExtension.version() });
+
+    // setup engine
+    const port = await this.verifyOrStartServerProcess({ ext, wsService });
+    Logger.info({ ctx: `${ctx}:verifyOrStartServerProcess`, port });
+    const engine = updateEngineAPI(port, ext);
+    Logger.info({ ctx: `${ctx}:exit` });
+
+    return { data: { workspace, engine, wsService } };
+  }
+
+  /**
+   * Initialize engine and activate workspace watchers
+   */
+  async activate({
+    ext,
+    context,
+    wsService,
+    wsRoot,
+    opts,
+    engine,
+  }: WorkspaceActivatorOpts &
+    WorkspaceActivatorSkipOpts & {
+      engine: EngineAPIService;
+      wsService: WorkspaceService;
+    }): Promise<RespV3<boolean>> {
+    const ctx = "WorkspaceActivator:activate";
+    // setup services
+    context.subscriptions.push(TextDocumentServiceFactory.create(ext));
+
+    // Setup tree viiew
+    const providerConstructor = function () {
+      return new EngineNoteProvider(engine);
+    };
+    if (!opts?.skipTreeView) {
+      await initTreeView({ context, providerConstructor });
+    }
+
+    // Reload
+    WSUtils.showActivateProgress();
+    const start = process.hrtime();
+    const reloadSuccess = await reloadWorkspace({ ext, wsService });
+    const durationReloadWorkspace = getDurationMilliseconds(start);
+
+    // NOTE: tracking is not awaited, don't block on this
+    ExtensionUtils.trackWorkspaceInit({
+      durationReloadWorkspace,
+      activatedSuccess: !!reloadSuccess,
+      ext,
+    });
+    analyzeWorkspace({ wsService });
+
+    if (!reloadSuccess) {
+      HistoryService.instance().add({
+        source: "extension",
+        action: "not_initialized",
+      });
+      return {
+        error: ErrorFactory.createInvalidStateError({
+          message: `issue with init`,
+        }),
+      };
+    }
+
+    ExtensionUtils.setWorkspaceContextOnActivate(wsService.config);
+    MetadataService.instance().setDendronWorkspaceActivated();
+    Logger.info({ ctx, msg: "fin startClient", durationReloadWorkspace });
+
+    const stage = getStage();
+    if (stage !== "test") {
+      ext.activateWatchers();
```

Comment:
This throws an error if the tutorial workspace is initialized in a non-workspace folder.

```
stack trace: Error: no folders set for workspace
	at DendronExtension.activateWatchers (/Users/hikchoi/dev/dendronhq/dendron/packages/plugin-core/out/src/workspace.js:555:19)
	at WorkspaceActivator.activate (/Users/hikchoi/dev/dendronhq/dendron/packages/plugin-core/out/src/workspace/workspaceActivator.js:421:17)
	at processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async SetupWorkspaceCommand.execute (/Users/hikchoi/dev/dendronhq/dendron/packages/plugin-core/out/src/commands/SetupWorkspace.js:227:21)
	at async LaunchTutorialWorkspaceCommand.execute (/Users/hikchoi/dev/dendronhq/dendron/packages/plugin-core/out/src/commands/LaunchTutorialWorkspaceCommand.js:34:9)
	at async LaunchTutorialWorkspaceCommand.run (/Users/hikchoi/dev/dendronhq/dendron/packages/plugin-core/out/src/commands/base.js:69:20)
	at async /Users/hikchoi/dev/dendronhq/dendron/packages/plugin-core/out/src/WelcomeUtils.js:52:21
```